beam-migrate/Database/Beam/Migrate/Backend.hs

```diff
@@ -73,6 +74,7 @@ type DdlError = String
 data BeamMigrationBackend be m where
   BeamMigrationBackend ::
     ( MonadBeam be m
+    , Fail.MonadFail m -- TODO check whether this is good
```

Comment:
Can you remove the TODO? I agree that it would be nice to have something more structured but I think this is correct for now.